helm/nessie/templates/_helpers.tpl

```diff
@@ -92,3 +92,133 @@ names should coincide with jdbc schemes in connection URIs.
 {{- $v := . | split ":" -}}
 {{ $v._1 }}
 {{- end }}
+
+{{/*
+Apply Iceberg catalog options.
+*/}}
+{{- define "nessie.applyCatalogIcebergOptions" -}}
+{{- $root := index . 0 -}}{{/* the object to introspect */}}
+{{- $map := index . 1 -}}{{/* the destination map */}}
+{{- with $root -}}
+{{- $_ := set $map "nessie.catalog.default-warehouse" .defaultWarehouse -}}
+{{- range $k, $v := .configDefaults -}}
+{{- $_ = set $map ( printf "nessie.catalog.iceberg-config-defaults.%s" $k ) $v -}}
+{{- end -}}
+{{- range $k, $v := .configOverrides -}}
+{{- $_ = set $map ( printf "nessie.catalog.iceberg-config-overrides.%s" $k ) $v -}}
+{{- end -}}
+{{- range $i, $warehouse := .warehouses -}}
+{{- if not $warehouse.name -}}{{- required ( printf "catalog.iceberg.warehouses[%d]: missing warehouse name" $i ) $warehouse.name -}}{{- end -}}
+{{- $_ = set $map ( printf "nessie.catalog.warehouses.%s.location" ( quote $warehouse.name ) ) $warehouse.location -}}
+{{- range $k, $v := $warehouse.configDefaults -}}
+{{- $_ = set $map ( printf "nessie.catalog.warehouses.%s.iceberg-config-defaults.%s" ( quote $warehouse.name ) $k ) $v -}}
+{{- end -}}
+{{- range $k, $v := $warehouse.configOverrides -}}
+{{- $_ = set $map ( printf "nessie.catalog.warehouses.%s.iceberg-config-overrides.%s" ( quote $warehouse.name ) $k ) $v -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Apply S3 catalog options.
+*/}}
+{{- define "nessie.applyCatalogStorageS3Options" -}}
```

Comment:
These templates apply to both the default config, and the per-bucket configs. But note that some options are only supported at the default level. If they are supplied at per-bucket level, the chart would install but the pod would fail to start with SmallRye complaining that the option is not mapped.

Reply:
The ones in these classes:

* `org.projectnessie.catalog.files.s3.S3Config`
* `org.projectnessie.catalog.files.adls.AdlsConfig`